internal/mode/static/telemetry/platform.go

```diff
@@ -0,0 +1,101 @@
+package telemetry
+
+import (
+	"strings"
+
+	v1 "k8s.io/api/core/v1"
+)
+
+type k8sState struct {
+	node       v1.Node
+	namespaces v1.NamespaceList
+}
+
+type platformExtractor func(k8sState) (string, bool)
+
+func buildProviderIDExtractor(id string, platform string) platformExtractor {
+	return func(state k8sState) (string, bool) {
+		if strings.HasPrefix(state.node.Spec.ProviderID, id) {
+			return platform, true
+		}
+		return "", false
```

Comment:
Do we need to return a bool here? Can we simplify and just check if the string is empty?